src/averageAges.js

```diff
@@ -15,11 +15,24 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let menAverageAges = 0;
+  let menAverageAges18 = 0;
+
+  menAverageAges18 = people
+    .filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century)
+    .map((p) => p.died - p.born);
+
+  menAverageAges = people
+    .filter((p) => p.sex === 'm')
+    .map((p) => p.died - p.born);
+
+  return (century === 18)
```

Comment:
The check for a century is only comparing with 18. The function is supposed to support any century, not just the 18th.